Test failure: separate_prod_fin_test

https://github.com/AntelopeIO/spring/actions/runs/10309295866/job/28539031748

```
2024-08-08T21:03:57.088340 {'qc': {'block_num': 151, 'active_policy_sig': {'strong_votes': '01111', 'sig': 'SIG_BLS_0VtwjvKLWTQyS-8nrdTQLghIue8POyp4kBHZRjLoooiflZryYJ6GvvQNEMhTCLAU3-LT88e9j8f1qsp7Q7kmhYN1F34LsLNmSJibug06utVppp9dxmqbGRM9USvEKBAAbcUf0sZ9FfMrN1HYWAz3cyGsAExM8mSAY3FhULV4w2PEx-CjEizUp6to6hU1BM8QGlxFU_VRNqrZJEchJKdEyUbVGlSAfipBX3ueJjFr_BKgfO_LJEV7eSQ4IjXza40UNFdBdg'}}}
2024-08-08T21:03:57.088536  Test failed.
2024-08-08T21:03:57.088734   Cluster shutting down.
Traceback (most recent call last):
  File "/__w/spring/spring/build/tests/separate_prod_fin_test.py", line 77, in <module>
    assert qc_ext["qc"]["active_policy_sig"]["strong_votes"] == "11111", 'Not all finalizers voted'
AssertionError: Not all finalizers voted
```

The node didn't get one of the votes in time. Seems like the test is too fragile. Maybe the test should check a few blocks and verify at least one of them has all strong votes.

--- a/tests/separate_prod_fin_test.py
+++ b/tests/separate_prod_fin_test.py
@@ -66,15 +66,21 @@ try:
     info = cluster.biosNode.getInfo(exitOnError=True)
     assert (info["head_block_num"] - info["last_irreversible_block_num"]) < 9, "Instant finality enabled LIB diff should be small"
 
-    # LIB has advanced, which indicate at least 2 of non-producer finalizers have voted.
     # Double check that's indeed the case in qc_extension
-    info = cluster.getNode(1).getInfo(exitOnError=True)
-    block_num = info["last_irreversible_block_num"]
-    block = cluster.getNode(1).getBlock(block_num)
-    qc_ext = block["qc_extension"]
-    Print(f'{qc_ext}')
-    # "11111" is the representation of a bitset showing which finalizers have voted (we have five total finalizers)
-    assert qc_ext["qc"]["active_policy_sig"]["strong_votes"] == "11111", 'Not all finalizers voted'
+    i = 0
+    while i < 3: # slow ci/cd might not get all votes in time, try a few blocks
+        # LIB has advanced, which indicate at least 2 of non-producer finalizers have voted.
+        block_num = cluster.getNode(1).getIrreversibleBlockNum()
+        block = cluster.getNode(1).getBlock(block_num)
+        qc_ext = block["qc_extension"]
+        Print(f'{qc_ext}')
+        # "11111" is the representation of a bitset showing which finalizers have voted (we have five total finalizers)
+        if qc_ext["qc"]["active_policy_sig"]["strong_votes"] == "11111":
+            break
+        cluster.getNode(1).waitForLibToAdvance()
+        i += 1
+    assert i < 3, "Not all finalizers voted"
+
 
     testSuccessful=True
 finally:

